app/models/patient_import.rb

```diff
@@ -128,6 +128,55 @@ def commit_changesets(changesets)
     end
   end
 
+  def destroy_parent_relationships_and_invalidate_consents!(
+    current_user,
+    consents
+  )
+    ActiveRecord::Base.transaction do
+      parent_relationships =
+        self.parent_relationships.includes(:parent).includes(:patient)
+
+      parents_to_check = parent_relationships.map(&:parent)
+
+      parent_relationships.destroy_all
+
+      parents_to_check.each do |parent|
+        if parent.parent_relationships.empty? && parent.consents.empty?
+          parent.destroy!
+        end
+      end
+
+      consents.each do |consent|
+        timestamp = Time.current.to_fs(:long)
+        user_name = current_user&.full_name || "system"
+        invalidation_note =
+          "Consent invalidated on #{timestamp} " \
+            "because #{user_name} removed all parent-child relationships from an import."
```

Comment:
These are the same for all the consents, can we calculate these first and then perform `update_all` to update all the consents in one go?